packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -425,18 +420,12 @@ export class FluidDataStoreRuntime
 	public async getChannel(id: string): Promise<IChannel> {
 		this.verifyNotClosed();
 
-		// TODO we don't assume any channels (even root) in the runtime. If you request a channel that doesn't exist
-		// we will never resolve the promise. May want a flag to getChannel that doesn't wait for the promise if
-		// it doesn't exist
-		if (!this.contextsDeferred.has(id)) {
-			this.contextsDeferred.set(id, new Deferred<IChannelContext>());
+		const context = this.contexts.get(id);
+		if (context === undefined) {
+			throw new LoggingError("Channel does not exist");
```

Comment:
Not sure if this is the most appropriate error type, but it seems comparable with other errors in this file.

Reply:
My instinct would be `UsageError`, which extends `LoggingError` anyway but is the one we use to indicate "this is not a valid use of this API" (which I quickly interpret as invalid arguments, called at the wrong time / with the wrong internal state, and similar).